docs/getting-started/example.md

```diff
@@ -65,11 +64,72 @@ export default function Component() {
 }
 ```
 
-:::tip
 You can use the `useSession` hook from anywhere in your application (e.g. in a header component).
-:::
 
-### Share/configure session state
+### Backend - API Route
+
+To protect an API Route, you can use the [`getSession()`](/getting-started/client#getsession) method in the NextAuth.js client.
+
+```javascript
+import { getSession } from "next-auth/react"
+
+export default async (req, res) => {
+  const session = await getSession({ req })
+
+  if (session) {
+    res.send({
+      content:
+        "This is protected content. You can access this content because you are signed in.",
+    })
+  } else {
+    res.send({
+      error: "You must be sign in to view the protected content on this page.",
+    })
+  }
+}
+```
+
+### Extensibility
+
+#### Using NextAuth.js Callbacks
+
+NextAuth.js allows you to hook into various parts of the authentication flow via our [built-in callbacks](/configuration/callbacks).
+
+For example, to pass a value from the sign-in to the frontend, client-side, you can use a combination of the [`session`](/configuration/callbacks#session-callback) and [`jwt`](/configuration/callbacks#jwt-callback) callback like so:
+
+```javascript
+...
+callbacks: {
+  async jwt({ token, account }) {
+    // Persist the OAuth access_token to the token right after signin
+    if (account) {
+      token.accessToken = account.access_token
+    }
+    return token
+  },
+  async session({ session, token, user }) {
+    // Send properties to the client, like an access_token from a provider.
+    session.accessToken = token.accessToken
+    return session
+  }
+}
+...
+```
+
+Now whenever you call `getSession` or `useSession`, the data object which is returned will include the `accessToken` value.
+
+```js
+import { useSession, signIn, signOut } from "next-auth/react"
+
+export default function Component() {
+  const { data } = useSession()
+  const { accessToken } = data
+
+  return <div>Access Token: {accessToken}</div>
+}
+```
+
+#### Share/configure session state
```

Comment:
This section must come before or at the beginning of "Frontend - Add React Hook", since it is essential for `useSession` to work.